### PR TITLE
Fix some lgtm convention alerts

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- The missing commas are actually bugs.
- `zs = [float(zs)] * len(xs)` is just rewritten to make the intention more clear. Can we make zs a numpy array instead of a list here? If so, that can be further simplified to `np.full_like(xs, fill_value=float(zs))`.

### CODE REVIEW

**Comment by timhoffm** [comment]
Any opinions on the `np.full_like`?

**Comment by jklymak** [comment]
Hah, I always do that z = x*0 + z

**Comment by dopplershift** [comment]
👍 to `np.full_like`

**Comment by timhoffm** [comment]
Changed to `np.full_like`

> Hah, I always do that z = x*0 + z

That's almost the same :smile: The results differ if x contains nan values. Also `x*0 + z` is slightly slower (1ms vs. 366µs for a size of 1 000 000), not that this would matter in the present context.

**Comment by anntzer** [comment]
@jklymak that one's cute

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -128,8 +128,8 @@
     "/Network/Library/Fonts/",
     "/System/Library/Fonts/",
     # fonts installed via MacPorts
-    "/opt/local/share/fonts"
-    ""
+    "/opt/local/share/fonts",
```
Comment (dsentinel) [inline lib/matplotlib/font_manager.py]: This comma changed semantics from implicit string concatenation to a new empty string in the list.

Comment (tacaswell) [reply lib/matplotlib/font_manager.py]: This is correcting a bug, we want to pick up fonts in the current directory.

Comment (dsentinel) [reply lib/matplotlib/font_manager.py]: Only on OSX, not windows or Linux?
Doesn't seem like something we'd want.

Comment (tacaswell) [reply lib/matplotlib/font_manager.py]: Hmm, that seems like a bug that it is missing in the other cases.

On the other hand, this recursion issue seems to be causing grief for many people and as this is cached,  using the cwd is also a bit off (as the next time you read the cache your cwd is different).

On balance, I think my initial reaction to this was wrong.

**Pull request merged**